log4j-layout-template-json/src/main/java/org/apache/logging/log4j/layout/template/json/resolver/ReadOnlyStringMapResolver.java

```diff
@@ -202,9 +202,7 @@ public void resolve(final LogEvent logEvent, final JsonWriter jsonWriter) {
                 final ReadOnlyStringMap map = mapAccessor.apply(logEvent);
                 final Object value = map == null ? null : map.getValue(key);
                 if (stringified) {
-                    final String valueString =
-                            ParameterizedMessage.deepToString(value);
-                    jsonWriter.writeString(valueString);
```

Comment:
When we moved from `Objects.toString(value)` -> `ParameterizedMessage.deepToString(value)` we also changed null handling, where null values previously resulted in the string `"null"` but `ParameterizedMessage.deepToString(` produces a null string in that case, so we would write a json null token.

writeAsString will write the value as `"null"` once again. I should add test coverage separately.